library/Icingadb/Web/Control/SearchBar/ObjectSuggestions.php

```diff
@@ -301,7 +301,7 @@ protected function queryCustomvarConfig(string $searchTerm): Select
         }
 
         $customVars->columns('flatname');
-        $this->applyBaseFilter($customVars);
+        $this->applyRestrictions($customVars);
```

Comment:
Please explain why this is required in such a global way. At the moment, this disables base filter appliance for all routes that have one and show a searchbar.

Reply:
e.g. hostgroup, servicegroup

Reply:
As discussed offline, the base filter is not necessary here as it is already applied to the inner query (`$scalarQueries`) with `applyBaseFIlter()`.
Yes, the `applyBaseFIlter()` also applies restrictions to the inner query, but the `denylist/protect` can only be applied correctly if the `flatname` column is fetched.
Since the outer query retrieves the `flatname` column, the restriction must also be applied here (introduced with https://github.com/Icinga/icingadb-web/commit/5a808ae6d2063fba8fd667b214ad4550eb7f08fe).